### PR TITLE
[fplus] Update to 0.2.18

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates fplus to 0.2.18

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

**Comment by Thomas1664** [comment]
Please remove unused cmake var:

```
CMake Warning at D:/installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:418 (message):
  The following variables are not used in CMakeLists.txt:

      FPLUS_BUILD_EXAMPLES
```

File: ports/fplus/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {
   "name": "fplus",
-  "version-semver": "0.2.14-p0",
+  "version-semver": "0.2.18-p0",
   "description": "Functional Programming Library for C++. Write concise and readable C++ code",
-  "homepage": "https://github.com/Dobiasd/FunctionalPlus"
+  "homepage": "https://github.com/Dobiasd/FunctionalPlus",
```
Comment (Thomas1664) [inline ports/fplus/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/Dobiasd/FunctionalPlus",
  "license": "BSL-1.0",
```

**Comment by chausner** [comment]
> Please remove unused cmake var:
> 
> ```
> CMake Warning at D:/installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:418 (message):
>   The following variables are not used in CMakeLists.txt:
> 
>       FPLUS_BUILD_EXAMPLES
> ```

Thanks, removed it.

File: ports/fplus/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,13 @@
 {
   "name": "fplus",
-  "version-semver": "0.2.14-p0",
+  "version-semver": "0.2.18-p0",
```
Comment (LilyWangLL) [inline ports/fplus/vcpkg.json:3]:
```suggestion
  "version": "0.2.18-p0",
```
Is this port a sermer library? If it is not, here hould use `"version"`?

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/fplus/vcpkg.json (hunk 2 above)
Comment (chausner) [reply ports/fplus/vcpkg.json:3]: In theory, the -p0 suffx is compliant with semver. @JonLiu1993 added the version-semver in https://github.com/microsoft/vcpkg/pull/16510 but I can't find any reference in the upstream repo that would indiciate that the project adheres to semver.

/cc @Dobiasd @JonLiu1993

Comment (JonLiu1993) [reply ports/fplus/vcpkg.json:3]: @chausner, I went back to investigate, the modification I made a year ago was not rigorous, and I think this update should change "version-semver" to "version"

Comment (Dobiasd) [reply ports/fplus/vcpkg.json:3]: @chausner Yeah, FunctionalPlus does not always strictly adhere to semver. :white_check_mark:

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**